### PR TITLE
Support for sourceGenerators in paradoxMarkdownToHtml

### PR DESCRIPTION
**Pull request by pyoio**
### Purpose

This PR introduces the ability to specify `sourceGenerators` for `paradoxMarkdownToHtml` allowing code to generate markdown pages to be rendered into the final result.

* Functionality implemented in `plugin/src/main/scala/com/lightbend/paradox/sbt/ParadoxPlugin.scala`
* Tests added under `plugin/src/sbt-test/paradox/generated-source`
* Documentation added under `docs/src/main/paradox/customization/generators.md`

### Example Usage:

```scala
lazy val root = (project in file(".")).
  enablePlugins(ParadoxPlugin).
  settings(
    name := "Generator Project",
    paradoxTheme := Some(builtinParadoxTheme("generic")),
    sourceGenerators in (Compile, paradoxMarkdownToHtml) += Def.task {
      val file = (sourceManaged in (Compile, paradox)).value / "generated-page.md"
    
      IO.write(file, "# Markdown generated by code!")
    
      Seq(file)
    }.taskValue
  )
```

And then you can reference `generated-page.md` from your non-generated sources under `src/main/paradox`.

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @pyoio,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by pyoio** [comment]
@lightbend-cla-validator done

**Comment by pyoio** [comment]
Rebased from master to fix tests (normalize.css no longer present in generated files)

File: plugin/src/sbt-test/paradox/generated-source/build.sbt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+lazy val docs = project
+  .in(file("."))
+  .enablePlugins(ParadoxPlugin)
+  .settings(
+    name := "Paradox Source Generation Test",
+    paradoxProperties += ("canonical.base_url" -> "https://example.com/doc/"),
+    paradoxRoots := List("index.html"),
```
Comment (pvlugter) [inline plugin/src/sbt-test/paradox/generated-source/build.sbt]: I'm assuming `paradoxProperties` and `paradoxRoots` are not needed for this test.

Comment (pvlugter) [reply plugin/src/sbt-test/paradox/generated-source/build.sbt]: Also, add a `paradoxTheme := None` setting to remove all default theme stuff from the expected output.

Add an empty template file `src/main/paradox/_template/page.st` with:

```
$page.content$
```

And then you'll just have the minimum expected files.

File: docs/src/main/paradox/customization/generators.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    paradoxTheme := Some(builtinParadoxTheme("generic")),
+    sourceGenerators in (Compile, paradoxMarkdownToHtml) += Def.task {
+      val file = (sourceManaged in (Compile, paradox)).value / "generated-page.md"
+    
+      IO.write(file, "# Markdown generated by code!")
+    
+      Seq(file)
+    }.taskValue
+  )
+```
```
Comment (pvlugter) [inline docs/src/main/paradox/customization/generators.md]: I'd like to see these as tested code snippets (in scripted test, pulled into the doc) at some point, like we do in Cinnamon and elsewhere. It looks like we don't have this for other examples like this in the paradox docs, so fine to leave this as code directly for now. (And if we do that we can still have the generic theme in the snippet and then override sbt setting to no theme to simplify output in testing.)

File: plugin/src/sbt-test/paradox/generated-source/build.sbt (hunk 1 above)
Comment (pyoio) [reply plugin/src/sbt-test/paradox/generated-source/build.sbt]: Done! Cheers

**Pull request merged**